components/Typography/src/MDCTypography.m

```diff
@@ -185,45 +186,135 @@ + (UIFont *)boldFontFromFont:(UIFont *)font {
 
 @end
 
+@interface MDCSystemFontLoader ()
+
+/**
+ Basic cache to keep track of recently-used fonts.
+ */
+@property(nonatomic, strong) NSCache *fontCache;
+
+@end
+
 @implementation MDCSystemFontLoader
 
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _fontCache = [[NSCache alloc] init];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(didChangeContentSizeCategory)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
+  }
+  return self;
+}
+
+- (void)didChangeContentSizeCategory {
+  [_fontCache removeAllObjects];
+}
+
 - (UIFont *)lightFontOfSize:(CGFloat)fontSize {
+  NSString *cacheKey = [NSString stringWithFormat:@"light-%06f", fontSize];
+  UIFont *font = [self.fontCache objectForKey:cacheKey];
+  if (font) {
+    return font;
+  }
+
   if ([UIFont respondsToSelector:@selector(systemFontOfSize:weight:)]) {
-    return [UIFont systemFontOfSize:fontSize weight:UIFontWeightLight];
+    font = [UIFont systemFontOfSize:fontSize weight:UIFontWeightLight];
+  } else {
+    font = [UIFont fontWithName:@"HelveticaNeue-Light" size:fontSize];
+  }
+
+  if (font) {
+    [self.fontCache setObject:font forKey:cacheKey];
   }
-  return [UIFont fontWithName:@"HelveticaNeue-Light" size:fontSize];
+  return font;
 }
 
 - (UIFont *)regularFontOfSize:(CGFloat)fontSize {
+  NSString *cacheKey = [NSString stringWithFormat:@"regular-%06f", fontSize];
```

Comment:
Inline cache keys make me nervous because it's difficult to enforce uniqueness. Have you considered using `_cmd` to generate the cache keys based on the method signature?